insights/parsers/horizon_local_settings_conf.py

```diff
@@ -0,0 +1,242 @@
+           and then the values are extracted and converted to the corresponding data types.
+        """
+        # No data found or file is empty
+        if len(data) == 0:
+            raise ParseException("Empty file")
+
+        self.settings = {}
+        # Check if line does not starts with lowercase letters
+        local_settings = "\n".join([setting for setting in get_active_lines(data) if not re.match(r"^[a-z]+", setting)])
+
+        # Check if line contains a key
+        keys = []
+        for setting in local_settings.split("\n"):
+            key = re.findall(r"^[A-Z_\[\]\"]*\s*=", setting)
+            keys.extend(key) if len(key) > 0 else None
+
+        # No valid keys are found
+        if len(keys) == 0:
+            raise ParseException("No valid keys found.")
+
+        # Extract the keys and corresponding values from data and store in a dictionary
+        for key in range(len(keys)):
+            start = local_settings.index(keys[key]) + len(keys[key]) + 1
+            d_key = keys[key].strip("=").strip()
+            # Value is not present for a key
+            if (key < len(keys) - 1 and start == local_settings.index(keys[key + 1])) or \
+                    (key + 1 == len(keys) and not local_settings[start:]):
+                raise SkipException("Value not present for the key {0}.".format(d_key))
+            newline_index = local_settings[start:].find("\n")
+            if newline_index != -1:
+                check = local_settings[start:start + newline_index]
+            else:
+                check = local_settings[start:start + local_settings[start:].find("\"", 2) + 1]
+            if re.match(r"^[A-Za-z0-9\"]", check.replace("'", "\"")):
+                self.settings[d_key] = check
+            else:
+                if key < len(keys) - 1:
+                    self.settings[d_key] = local_settings[start:local_settings.index(keys[key + 1])]
+                else:
+                    self.settings[d_key] = local_settings[start:len(local_settings) + 1]
+            self.settings[d_key] = self.settings[d_key].replace("'", "\"").strip("\n")
+
+        # Convert the values from String to corresponding data types
+        for k, v in self.settings.items():
+            # Boolean or Constant or Numbers to String
+            if v[0].isalnum():
+                self.settings[k] = str(v)
+            # String
+            elif v.startswith("\""):
+                self.settings[k] = str(v[1:-1])
+            # List or Tuple to List
+            elif v.startswith("[") or v.startswith("("):
+                v = v[1:-1].replace("\"", "").split(",")
+                for elem in v:
+                    v[v.index(elem)] = elem.strip() if elem.strip("\n").startswith(" ") else elem.strip("\n")
+                self.settings[k] = v
+            # Dictionary
+            elif v.startswith("{"):
+                v = v + "}" if not v.endswith("}") else v
```

Comment:
This could be correct by modifying the filter. (e.g. add the "{" and "}")

Reply:
Acknowledged. Changes done in the latest commit.